src/Tribe/Utils/Element_Classes.php

```diff
@@ -0,0 +1,238 @@
+		return $this->get_attribute();
+	}
+
+	/**
+	 * Gets the full HTML class attribute for this instance of Element Classes.
+	 * It will contain a space on each end of the attribute.
+	 *
+	 * @since  TBD
+	 *
+	 * @return string In the format ` class="class1 class2" `
+	 */
+	public function get_attribute() {
+		$classes = $this->get_classes_as_string();
+		return " class=\"{$classes}\" ";
+	}
+
+	/**
+	 * Gets a space separated string of all classes to be printed.
+	 *
+	 * @since  TBD
+	 *
+	 * @return string
+	 */
+	public function get_classes_as_string() {
+		return implode( ' ', $this->get_classes() );
+	}
+
+	/**
+	 * Get the array of classes to be printed.
+	 *
+	 * @since  TBD
+	 *
+	 * @return array
+	 */
+	public function get_classes() {
+		$this->results = [];
+		$classes       = [];
+
+		$this->parse_array( $this->arguments );
+
+		foreach ( $this->results as $key => $val ) {
+			if ( ! $val ) {
+				continue;
+			}
+
+			$classes[] = $key;
+		}
+
+		$classes = array_map( 'sanitize_html_class', $classes );
+		$classes = array_filter( array_unique( $classes ) );
+
+		return $classes;
+	}
+
+
+	/**
+	 * Get the array of the classes, using [ class_name => bool ] as the format.
+	 *
+	 * @since  TBD
+	 *
+	 * @return array [ class_name => bool ]
+	 */
+	public function get_conditions() {
+		$this->results = [];
+		$this->parse_array( $this->arguments );
+
+		return $this->results;
+	}
+
+	/**
+	 * Parse arguments or argument for this instance, and store values on results.
+	 *
+	 * @since  TBD
+	 *
+	 * @param  mixed  $arguments      Any possible set of arguments that this class supports.
+	 * @param  boolean $default_value What is the default value for a given class.
+	 *
+	 * @return void
+	 */
+	protected function parse( $arguments, $default_value = true ) {
+		if ( ! $arguments ) {
+			return;
+		}
+
+		if ( is_string( $arguments ) ) {
+			// 'foo bar'
+			$this->parse_string( $arguments );
+		} elseif ( $arguments instanceof \Closure || is_callable( $arguments ) ) {
+			// function() {}
+			$this->parse_callable( $arguments );
+		} elseif ( is_array( $arguments ) ) {
+			// ['foo', 'bar', ...] || ['foo' => TRUE, 'bar' => FALSE, ...]
+			$this->parse_array( $arguments );
+		} elseif ( is_object( $arguments ) ) {
+			// stdClass
+			$this->parse_object( $arguments );
+		} elseif ( is_int( $arguments ) ) {
+			// '130'
+			$this->results[ $arguments ] = $default_value;
+		} elseif ( is_float( $arguments ) ) {
+			$this->results[ (string) $arguments ] = $default_value;
+		}
+	}
+
+	/**
+	 * Parse a string into an array of acceptable values for the instance.
+	 *
+	 * @since  TBD
+	 *
+	 * @param  string  $arguments     Space separated string of classes to be parsed
+	 * @param  boolean $default_value What is the default value for a given class.
+	 *
+	 * @return void
+	 */
+	protected function parse_string( $arguments, $default_value = true ) {
+		$values = preg_split( '/\s+/', $arguments, -1, PREG_SPLIT_NO_EMPTY );
```

Comment:
`preg_split` might return `false` on failure so `$values` will trigger an error in that case as `false` is not an array